src/pathfinder/opf/opf_ship.cpp

```diff
@@ -180,7 +187,7 @@ bad:;
 	} while (bits != TRACK_BIT_NONE);
 
 	*track = best_track;
-	return best_bird_dist;
+	return depottile != INVALID_TILE ? best_length : best_bird_dist;
```

Comment:
Changing the meaning of the return value based on the presence/absence of a parameter that sounds like it has nothing to do with that is a bad smell. Change the function back to always return the same, but take an optional pointer argument, where the caller can point to somewhere they want the travel distance stored, in addition to getting the direct distance as return value.

Reply:
I'm changing it to return best_length always, which is different than what it was on master.
OPFShipChooseTrack will be getting different values from it, but at a glance, the resulting behaviour of that function won't change.
OPFShipFindNearestDepot is the other function that uses FindShipTrack, and was created now with this patch, specially tailored towards best_length being returned.

OPFShipChooseTrack and OPFShipFindNearestDepot are the only functions that call FindShipTrack.